staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json.go

```diff
@@ -119,11 +147,28 @@ func CaseSensitiveJsonIterator() jsoniter.API {
 	return config
 }
 
-// Private copy of jsoniter to try to shield against possible mutations
+// StrictCaseSensitiveJsonIterator returns a jsoniterator API that's configured to be
+// case-sensitive, but also disallows unknown fields when unmarshalling. It is compatible with
+// the encoding/json standard library.
+func StrictCaseSensitiveJsonIterator() jsoniter.API {
```

Comment:
Consider taking a jsoniter.Config as an argument, setting the values we know we always need / clearing ones we know we don want. I think this library should provide tools that people can assemble as they wish.

(We can only certify it works in the configurations that we test it in, of course.)

Reply:
i'm little misunderstand it. it may be make the struct more complex